### PR TITLE
support use of listen options

### PR DESCRIPTION
**Pull request by Super-zhw**
Implementation of this https://github.com/asim/go-micro/issues/2535

### CODE REVIEW

**Comment by Super-zhw** [comment]
@asim 
Please review the code, if there is something wrong, please give me more advice.
If there is no problem, I hope to be able to join as soon as possible, thank you very much!

File: service_test.go [hunk 1]
```diff
@@ -3,6 +3,9 @@ package micro
 import (
 	"context"
 	"errors"
+	"go-micro.dev/v4/server"
```
Comment (asim) [inline service_test.go]: external imports should be separated from stdlib, please see other files as an example

**Comment by asim** [comment]
@xpunch please review

File: service_test.go (hunk 1 above)
Comment (Super-zhw) [reply service_test.go]: I'm sorry it was my oversight.

**Comment by Super-zhw** [comment]
@xpunch Hello, can you help me review this PR?

**Comment by asim** [comment]
@Super-zhw I can see you work at Tencent. Are you using this there?

**Comment by Super-zhw** [comment]
@asim 
At present, a small project in our team uses this component. Our team is relatively small and has limited capabilities. Thank you very much for opening up this component.

**Pull request merged**